packages/core/src/App/Containers/Layout/header/trading-hub-header.jsx

```diff
@@ -49,18 +49,19 @@ const RedirectToOldInterface = ({
     toggleExitTradersHubModal,
     content_flag,
     switchToCRAccount,
-    setPreferredLanguage,
 }) => {
-    const language = getLanguage();
+    const history = useHistory();
     const disablePreAppstore = async () => {
         if (should_show_exit_traders_modal) {
             toggleExitTradersHubModal();
         } else {
             if (content_flag === ContentFlag.LOW_RISK_CR_EU) {
                 await switchToCRAccount();
             }
-            setPreferredLanguage(language);
-            setIsPreAppStore(false);
+            history.push(routes.root);
+            setTimeout(() => {
+                setIsPreAppStore(false);
+            }, 800);
```

Comment:
Why setTimeout?

Reply:
`setIsPreAppstore` in the file is supposed to be async but if i put in await it will break

Reply:
@yashim-deriv can you please give your input

Reply:
This is an async call and we are also adding a `setTimeout` which delays `setIsPreAppStore` by creating a new execution context.